### DISCUSSION TITLE
Password-protected user interface

### DISCUSSION THREAD

**Discussion by pcmihnea** [jomjol/AI-on-the-edge-device#711] [Ideas]
Hi,

I'm currently using your awesome project on multiple meters around my home.
Unfortunately, the main wireless router is a home-user device that doesn't allow complex configurations. As such, almost all wireless devices are connected to the same network - this may prove unsecure, as the web interface is not protected by any means (password, HTTPS, access control, etc.) - all user settings (except WiFi configuration) are in plain-text form (MQTT credentials) and can be modified anytime.
Would it be possible to add a layer of protection to prevent access to the Web services deployed by your software?

Thank you!

**Comment by jomjol** [comment]
This is not the first request for a authenification. It is not planned to implement this in the free version.
Best regards,
  jomjol

**Reply by pcmihnea** [reply]
Ok, understood!

**Reply by jwillmer** [reply]
Is there a time plan for the paid version? It is really not good practice to keep it accessible. Maybe provide a simple Http basic auth in the meantime?

**Comment by jomjol** [comment]
No, there is no plan for a user paid version.
In a paid version there would be an expectation on ongoing maintenance, safety and security support, which cannot be maintained for an acceptable price.
There are some special commerical versions for industrial usage. But in this case the effort for changes are paid by effort directly.

**Reply by jwillmer** [reply]
Could we then include basic Auth to the project? Depending on the website it is a simple .htaccess file with 3 lines of code to make it happen. An example of other implementation options can be found at: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication

**Comment by friedpa** [comment]
Long time ago, it was decided in the AIotE Project Community not to include any Auth for the "free of charge" version.
If someone needs Auth, he can dircetly discuss that with @jomjol for a paid version of his (industrial) AIotE SW.

**Reply by tutsde** [reply]
> Long time ago, it was decided in the AIotE Project Community not to include any Auth for the "free of charge" version. If someone needs Auth, he can dircetly discuss that with @jomjol for a paid version of his (industrial) AIotE SW.

Sorry for Reply to this old Posting. Things can Change over time ....
Today I think a basic authentication is not (more) an indication for "industrial" produkt, but rather a minimum basic protection of every network-capable device. I am not a programming expert, but if it is easy to implement, it might at least be worth a (renewed) discussion among the developers...

**Reply by SybexX** [reply]
https://github.com/jomjol/AI-on-the-edge-device/pull/3436
